doc/connectivity/bluetooth/bluetooth-ctlr-arch.rst

```diff
@@ -0,0 +1,261 @@
+
+.. image:: img/ctlr_exec_lll_resume_bottom.png
+
+
+Bare metal utilities
+********************
+
+Memory FIFO and Memory Queue
+============================
+
+.. image:: img/ctlr_mfifo_memq.png
+
+
+Legacy Controller
+*****************
+
+.. image:: img/ctlr_legacy.png
+
+Bluetooth Low Energy Controller - Vendor Specific Details
+*********************************************************
+
+Hardware Requirements
+=====================
+
+Nordic Semiconductor
+--------------------
+
+The Nordic Semiconductor Bluetooth Low Energy Controller implementation
+requires the following hardware peripherals.
+
+.. list-table:: SoC Peripheral Use
+   :header-rows: 1
+
+   * - Resource
+     - nRF Peripheral
+     - Instances
+     - Zephyr Driver Accessible
+     - Description
+   * - Clock
+     - NRF_CLOCK
+     - 1
+     - Yes
+     - * A Low Frequency Clock (LFCLOCK) or sleep clock, for low power
+         consumption between Bluetooth radio events
+       * A High Frequency Clock (HFCLOCK) or active clock, for high precision
+         packet timing and software based transceiver state switching with
+         inter-frame space (tIFS) timing inside Bluetooth radio events
+   * - Real Time Counter (RTC)
+     - NRF_RTC0
+     - 1
+     - **No**
+     - * Uses 2 capture/compare registers
+   * - Timer
+     - NRF_TIMER0 or NRF_TIMER4 [1]_, and NRF_TIMER1 [0]_
+     - 2 or 1 [0]_
+     - **No**
+     - * 2 instances, one each for packet timing and tIFS software switching,
+         respectively
+       * 7 capture/compare registers (3 mandatory, 1 optional for ISR profiling,
+         4 for single timer tIFS switching) on first instance
+       * 4 capture/compare registers for second instance, if single tIFS timer
+         is not used.
+   * - Programmable Peripheral Interconnect (PPI)
+     - NRF_PPI
+     - 21 channels (20 [2]_), and 2 channel groups [3]_
+     - Yes [4]_
+     - * Used for radio mode switching to achieve tIFS timings, for PA/LNA
+         control
+   * - Distributed Programmable Peripheral Interconnect (DPPI)
+     - NRF_DPPI
+     -  20 channels, and 2 channel groups [3]_
+     - Yes [4]_
+     - * Used for radio mode switching to achieve tIFS timings, for PA/LNA
+         control
+   * - Software Interrupt (SWI)
+     - NRF_SWI4 and NRF_SWI5, or NRF_SWI2 and NRF_SWI3 [5]_
+     - 2
+     - **No**
+     - * 2 instances, for Lower Link Layer and Upper Link Layer Low priority
+         execution context
+   * - Radio
+     - NRF_RADIO
+     - 1
+     - **No**
+     - * 2.4 GHz radio transceiver with multiple radio standards such as 1 Mbps,
+         2 Mbps and Coded PHY S2/S8 Long Range Bluetooth Low Energy technology
+   * - Random Number Generator (RNG)
+     - NRF_RNG
+     - 1
+     - Yes
+     -
+   * - AES Electronic Codebook Mode Encryption (ECB)
+     - NRF_ECB
+     - 1
+     - **No**
+     -
+   * - Cipher Block Chaining (CBC) - Message Authentication Code with Counter
+       Mode encryption (CCM)
```

Comment:
This particular line makes the table very wide and quite difficult to see. You also have to scroll sideways to see more than just the first 2 columns. The scrollbar is also automatically hidden unless you mouseover the table, so unless you know it's there, you won't find it.